xbmc/addons/AddonEvents.h

```diff
@@ -48,6 +48,28 @@ namespace ADDON
       MetadataChanged(std::string id) : id(std::move(id)) {}
     };
 
+    /**
+     * Emitted when a different version of the add-on has been installed
+     * to the file system and should be reloaded.
+     */
+    struct ReInstalled: AddonEvent
+    {
+      std::string id;
+      ReInstalled(std::string id) : id(std::move(id)) {}
+    };
+
+    /**
+     * Emitted after the add-on has been uninstalled.
+     */
+    struct UnInstalled : AddonEvent
+    {
+      std::string id;
+      UnInstalled(std::string id) : id(std::move(id)) {}
```

Comment:
> Yes but if you have a running dll and this becomes deleted or changed during his runtime does Kodi crash.

Not on *nix, as long as it's kept open. I don't know what windows does (probably something stupid) but if this is an issue there, we can still handle it just fine in a non-blocking way (e.g by 'uninstalling' it from the addon system, but not actually delete the underlying files until later)